### ISSUE TITLE
8bitdo support

### ISSUE THREAD

**Issue by Simpleguyxorpad** [isage/vixen#2]
The controllers themselves work fine with this pluggin, but they randomly stop working but are still connected to pstv, and I was wondering if this could be fixed

**Comment by isage** [comment]
You'd have to be more specific

**Comment by Simpleguyxorpad** [comment]
I have 2 8bitdo controllers, both of them work perfectly in xinput mode but usually they disconnect randomly, they can go 1 hour without disconnecting or they can disconnect several times by themselves, I think it can be something in the pluggin because this controller has lights and I can assure you that they are still connected but pstv stops recognizing the inputs, if I can somehow send you something about this controller to fix it let me know

**Comment by isage** [comment]
Any chance you have "turn off controllers" option enabled in settings->power saving?

**Comment by Simpleguyxorpad** [comment]
It was the first thing I checked when I started having that problem but the options are disabled.

**Comment by isage** [comment]
[vixen.zip](https://github.com/isage/vixen/files/11048239/vixen.zip)

Install that version, wait until controller disconnects, find `ux0:/data/vixen.txt` and attach it

**Comment by Simpleguyxorpad** [comment]
[vixen.txt](https://github.com/isage/vixen/files/11057796/vixen.txt)
 After a few hours doing speedrun it disconnected and killed the run, I hope the sacrifice can fix the problem XD 

When the controller disconnected the game started running very slow until I disconnected It, I guess it's because of the version of vixen youbpost here to find the error

**Comment by isage** [comment]
Hm. Can you check on pc what usb vendor_id/product_id that controller has?
you can use https://gamepad-tester.com/ for that

**Comment by isage** [comment]
[vixen.zip](https://github.com/isage/vixen/files/11059350/vixen.zip)
And another test version. Same as above, run until it fails, send the file (remove the file if it exists prior to that)
Or, if it doesn't fail for quite some time, still check if file exists and send it.
This is really strange that it starts to spam errors on usb_write, since this is used for rumble/
led/power off.

**Comment by Simpleguyxorpad** [comment]
I am on the Gamepad tester page, where am I supposed to find the vendor and product id?  I will try to send the text file  as soon as possible.   thanks for the help

**Comment by isage** [comment]
![image](https://user-images.githubusercontent.com/273697/227652359-393ef52d-53cc-40fc-ae9b-d2f8309c6ee3.png)

**Comment by Simpleguyxorpad** [comment]
Ok while my controller was in xinput mode it only told me it was like a standard xbox controller, but if I change it to D-input it gives me this: 
8BitDo SN30 Pro+ (Vendor: 2dc8 Product: 6002) 
 And if I switch it to MAC-OS mode I get this 
 Sony PS4 Dualshock
Wireless Controller (STANDARD GAMEPAD Vendor: 054c Product: 05c4) 
but in xinput mode I only get this: 
Xbox 360 Controller (XInput STANDARD GAMEPAD)

**Comment by isage** [comment]
Oh well, i guess it just reports as standard xbox360 pad

**Comment by Simpleguyxorpad** [comment]
Do you think it would be a good idea or if it would be possible to map any of the other modes that comes with the controller to see if it also continues to happen in the other modes? (Dimput MACos switch) Maybe I won't have the same problem

**Comment by isage** [comment]
Maybe dinput, but let's try to figure out x-input first

**Comment by Simpleguyxorpad** [comment]
I can't use the latest vixen you sent me, it slows down ps1 games

**Comment by isage** [comment]
Huh. Check file, send it

**Comment by Simpleguyxorpad** [comment]
[vixen-1.txt](https://github.com/isage/vixen/files/11067123/vixen-1.txt)

**Comment by isage** [comment]
Okay, that's good. I'll cook another test tomorrow

**Comment by isage** [comment]
[vixen.zip](https://github.com/isage/vixen/files/11068298/vixen.zip)

**Comment by Simpleguyxorpad** [comment]
[vixen-2.txt](https://github.com/isage/vixen/files/11070495/vixen-2.txt)

**Comment by isage** [comment]
Still stops working?
Also, does it have rumble support?

**Comment by Simpleguyxorpad** [comment]
the problem persists, and the controller does have Rumble suport .Do you think it could be a problem with adrenaline or PS1 games?

**Comment by isage** [comment]
Does rumble work? Usually pstv can't provide enough power for rumble without usb hub with external power.
Also, try disabling vibration in quick settings (hold PS) and see if that helps.

**Comment by Simpleguyxorpad** [comment]
The Rumble works with the pluggin, tonight I will try disabling the Rumble and I might also try using a powered hub. If there is any news I will let you know here 👍.

**Comment by Simpleguyxorpad** [comment]
[vixen.txt](https://github.com/isage/vixen/files/11082971/vixen.txt)
It keeps happening even with vibration disabled and powered hub :/

**Comment by isage** [comment]
Welp, I'm out of ideas.
Did disabling vibration actually disabled it? 
Did you remove vixen.txt prior to testing?
Because log says it hangs on write, and write is only used for vibration.

We can try implementing dinput mode and see if that helps.

**Comment by Simpleguyxorpad** [comment]
the vibration stopped working when I deactivated it, I will use the new tool you developed and send it to you.

**Comment by Simpleguyxorpad** [comment]
When I try to map the imputs in dimput mode it does not work I get a popup of this control is not supported by psvita, but trying to take the maps in xinput/switch mode I noticed that it detects 2 devices, is that normal ?

**Comment by isage** [comment]
Popup is fine. Does the app see the device?

**Comment by Simpleguyxorpad** [comment]
no, with dinput mode it does not go beyond the press x screen.  Being in macOS mode, I see that it is connected because the light stays on but not in dinput mode, and in neither of the two let me get to see the vendor id  I can not pass from the screen that asks you to press x with the controller

**Comment by isage** [comment]
You need to press X on paired ds3/ds4, not on 8bitdo controller

**Comment by Simpleguyxorpad** [comment]
[vixen.txt](https://github.com/isage/vixen/files/11093655/vixen.txt)

I think the file is not right, it should also add the imputs but I think it does not register them.

**Comment by isage** [comment]
Yeah, it can't even read first report. Well, i'm out of ideas then.

**Comment by Simpleguyxorpad** [comment]
Anyway thanks for your help, maybe in the future the error can be found and fixed!
At the moment I'm waiting for a ps2usb adapter to at least use a ps2 controller.
